qucs-core/src/components/indq.cpp

```diff
@@ -43,29 +43,27 @@ indq::indq () : circuit (2) {
 // This function calculates the ABCD matrix of a lossy inductance
 // Q = 2*pi*f*L/Rs
 // where Rs is the series resistance and L the inductance
-void indq::calcABCDparams(nr_double_t frequency)
-{
+void indq::calcSP (nr_double_t frequency) {
  nr_double_t L = getPropertyDouble ("L");
  nr_double_t Q = getPropertyDouble ("Q");
  nr_double_t f = getPropertyDouble ("f");
- if ((f==0) || (Q==0)) Q = 1e10; //In case of unphysical inputs, the capacitor is forced to be ideal
- nr_double_t Qf=Q;
-
- if (!strcmp (getPropertyString ("Mode"), "Linear")) Qf*=frequency/f;
- if (!strcmp (getPropertyString ("Mode"), "Quadratic"))Qf*=qucs::sqrt(frequency/f);
- nr_double_t Rs = 1/(2*pi*frequency*L*Qf);
-
- ABCD = eye(2);
- ABCD.set(0,0,1);
- ABCD.set(0,1, Rs + 2*pi*L*frequency*nr_complex_t (0, 1));
- ABCD.set(1,0,0);
- ABCD.set(1,1,1);
-}
-
-void indq::calcSP (nr_double_t frequency) {
-  calcABCDparams(frequency);
-  matrix Stmp = qucs::atos(ABCD, z0, z0);
-  setMatrixS(Stmp);
+ nr_double_t Rs = 0;
+ if ((f!=0) && (Q!=0))
+ {
+   nr_double_t Qf=Q;
+   if (!strcmp (getPropertyString ("Mode"), "Linear")) Qf*=frequency/f;
+   if (!strcmp (getPropertyString ("Mode"), "Quadratic"))Qf*=qucs::sqrt(frequency/f);
+   Rs = 1./(2.*pi*frequency*L*Qf);
```

Comment:
oops, the `Rs` formula is wrong here, you got it right in the previous commit comment, though :grin: